sourcegraph/sourcegraph
# feat(setup-checklist): add checklist validity checking & redirecting base

Closes https://github.com/sourcegraph/sourcegraph/issues/55434.

This PR:
- Removes setup checklist integration with SetupWizard, because we for now decided that we will be having different UI approach
- Updates `ChecklistNavbarItem` to redirect to pages with additional tracking query parameters
- Adds `ChecklistInfo` component that parses query parameters to render step info if needed
- Update `useSetupChecklist` with template logic of calculating whether the license is set or code host connections are configured
   > NOTE: the actual checking logic of each items will probably be more sophisticated, such as checking that codehost connection is actually valid and can sync repos or that license key is not expiring and current users don't exceed allowed by licence count.

## Test plan
- `sg start`
- Enable `setup-checklist` feature flag
- Check there is a Navbar item with license & code host connections items
- Try removing codehosts or licenseKey and check navbar picks up changes and shows correct status

## Demo video

https://github.com/sourcegraph/sourcegraph/assets/6717049/988a43d2-6512-48a5-b970-e84c79777d6e
